### PR TITLE
Add workflow test to M. tuberculosis variant analysis tutorial

### PR DESCRIPTION
**Pull request by pvanheus**
This is a continuation of #2004.

### CODE REVIEW

**Comment by bgruening** [comment]
@cat-bro do you have time to look at this?

**Comment by cat-bro** [comment]
In the test each output should be listed only once, so
```
    mtbva_tb_variant_report_html:
      asserts:
        has_text:
          text: 'c.481C>T'
    mtbva_tb_variant_dr_report_html:
      asserts:
        has_text:
          text: 'Ser315Thr'
```
needs to be replaced with 
```
    mtbva_tb_variant_report_html:
      asserts:
        has_text:
          text: 'c.481C>T'
        has_text:
          text: 'Ser315Thr'
```

**Comment by cat-bro** [comment]
There is also an error in the final output test for `mtbva_jbrowse`.  Looks like the html page where the text is being searched for is the loading page.  History is shared here: https://usegalaxy.eu/u/cat/h/pvhwfga-1595721294 . This error is for the output at step 26.

```
Output with path /var/folders/4p/_rp9rd655ll4y21dmw2797j40000gn/T/tmpm7uxyonq/JBrowse on data 4, data 22, and others - Complete different than expected
Output file did not contain expected text 'Mycobacterium_tuberculosis_h37rv' (output '<!DOCTYPE html>
<html>
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
    <title>JBrowse</title>

    <link rel="apple-touch-icon" sizes="180x180" href="img/favicons/apple-touch-icon.png">
    <link rel="icon" type="image/png" sizes="32x32" href="img/favicons/favicon-32x32.png">
    <link rel="icon" type="image/png" sizes="16x16" href="img/favicons/favicon-16x16.png">
    <link rel="manifest" href="site.webmanifest">
    <link rel="mask-icon" href="img/favicons/safari-pinned-tab.svg" color="#5bbad5">
    <meta name="msapplication-TileColor" content="#2d89ef">
    <meta name="theme-color" content="#ffffff">

    <script type="text/javascript">
        window.onerror=function(msg){
            if( document.body )
                document.body.setAttribute("JSError",msg);
        }
        if(window.process&&process.versions&&process.versions.electron) {
            window.electronRequire = require;
            delete window.require;
        }
    </script>
    <style>
        html, body, div.jbrowse {
            margin: 0;
            padding: 0;
            height: 100%;
            width: 100%;
        }
    </style>
    <script type="text/javascript" src="dist/main.bundle.js" charset="utf-8"></script>
  </head>

  <body>
    <div class="jbrowse" id="GenomeBrowser" data-config='"allowCrossOriginDataRoot": false, "cacheBuster": true'>
      <div id="LoadingScreen" style="padding: 50px;">
        <h1>Loading...</h1>
      </div>
    </div>
    <div style="display: none">JBrowseDefaultMainPage</div>
  </body>
</html>
')
```

**Comment by pvanheus** [comment]
> `planemo test --galaxy_url https://usegalaxy.eu --galaxy_user_key <api_key> --no_shed_install --engine external_galaxy <path_to_workflow>`

Thanks, I tried that but get an error: `'GalaxyInstance' object has no attribute '_make_url'`.

I made the change to the test that you suggested. Indeed the problem here is that the JBrowse output is multiple files, so a simple test for '<html>' is all one can do.

**Comment by nsoranzo** [comment]
@pvanheus Try to update planemo, that error should be fixed in the latest version.

**Comment by pvanheus** [comment]
@hexylena development of this tutorial is 

> @pvanheus Try to update planemo, that error should be fixed in the latest version.

Thanks - I've updated to planemo 0.70 and it works (although curiously seems to insist on Python 2).

**Comment by pvanheus** [comment]
> @pvanheus Try to update planemo, that error should be fixed in the latest version.

Thanks - I've updated to planemo 0.70 and it works (although curiously seems to insist on Python 2).

File: topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.md [hunk 1]
```diff
@@ -131,13 +150,15 @@ tutorial on ["Quality control"]({% link topics/sequence-analysis/tutorials/quali
...
+*Note:* We would normally examine our trimmed reads with **FastQC** and **MultiQC** again to see if the quality trimming has been successful, but in this tutorial we will move straight on to save time.
+
 # Look for contamination with Kraken2
 
 We should also look for contamination in our reads. Sometimes, other sources of DNA accidentally or inadvertantly get mixed in with our sample. Any reads from non-sample sources will confound our snp analysis. **Kraken 2** is an effective way of looking and which species is represented in our reads and so we can easily spot possible contamination of our sample.
 
 > ### {% icon hands_on %} Hands-on: Run Kraken2
 >
-> 1. **Kraken2** {% icon tool %} with the following parameters
+> 1. Execute {% tool [Kraken2](tool_id=toolshed.g2.bx.psu.edu/repos/iuc/kraken2/kraken2/2.0.8_beta+galaxy0) %} {% icon tool %} with the following parameters
```
Comment (hexylena) [inline topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.md:161]:
```suggestion
> 1. Execute {% tool [Kraken2](toolshed.g2.bx.psu.edu/repos/iuc/kraken2/kraken2/2.0.8_beta+galaxy0) %} {% icon tool %} with the following parameters
```

**Comment by pvanheus** [comment]
I think this is ready for merging now. :)

**Comment by bgruening** [comment]
I will defer to @cat-bro here :)

**Comment by cat-bro** [comment]
yes, ready to merge :)

**Comment by bgruening** [comment]
Thanks everyone!

**Pull request merged**